proposals/NNNN-inline-spirv-attributes.md

```diff
@@ -0,0 +1,126 @@
+<!-- {% raw %} -->
+
+# Inline SPIR-V attributes
+
+## Instructions
+
+> This template wraps at 80-columns. You don't need to match that wrapping, but
+> having some consistent column wrapping makes it easier to view diffs on
+> GitHub's review UI. Please wrap your lines to make it easier to review.
+
+> When filling out the template below for a new feature proposal, please do the
+> following first:
+
+> 1. Copy this template to proposals and name it `NNNN-<some unique name>.md`.
+>    Do not assign the proposal a number.
+> 2. Exclude the "Planned Version", "PRs" and "Issues" from the header as
+>    appropriate.
+> 3. Fill out the "Introduction", "Motivation", and "Proposed Solution" sections
+>    of the template. Do not spend time writing the "Detailed design" until the
+>    initial proposal is merged as details will change through design iteration.
+> 4. Delete this Instructions section including the line below.
+> 5. Post your new proposal for review in a PR and work through review feedback.
+> 6. When your PR is approved, _immediately before merging_ update the PR to
+>    assign the proposal the highest unused proposal number. Giving each merged
+>    proposal a unique number allows it to be easily referred to in issues and
+>    project planning.
+
+---
+
+* Proposal: [NNNN](NNNN-filename.md)
+* Author(s): [Steven Perron](https://github.com/s-perron)
+* Status: **Design In Progress**
+
+*During the review process, add the following fields as needed:*
+
+* PRs: [#NNNN](https://github.com/llvm/llvm-project/pull/NNNN)
+* Issues:
+  [#NNNN](https://github.com/llvm/llvm-project/issues/NNNN)
+* Posts: [LLVM Discourse](https://discourse.llvm.org/)
+
+## Introduction
+
+SPIR-V has some extra information in the module that is required to generate
+correct code. These are decorations, capabilities, and extensions. They do not
+always match closely with existing llvm-ir attributes.
+
+## Motivation
+
+The inline SPIR-V feature defines 10 attributes that attach to function, function parameters, variables,
+types, and member variables:
+
+```HLSL
+vk::ext_instruction(uint opcode, [string extended_instruction_set])
+vk::ext_execution_mode(uint execution_mode, ...)
```

Comment:
This is the function that we want to deprecate. Replace with `vk::spvexecutionmode`.